From 2dc986fcaf51aff1e307df469da8feeb86574111 Mon Sep 17 00:00:00 2001
From: Francois Gouget <fgouget@codeweavers.com>
Date: Thu, 1 Dec 2016 05:19:33 +0100
Subject: [PATCH 15/55] udscs: Fix a potential NULL pointer dereference

udscs_server_fill_fds() should accept being passed a NULL pointer.
udscs_server_fill_fds() is dereferencing the 'server' pointer, and then
checks if it's NULL. This commit makes sure the NULL check happens
first.

Signed-off-by: Francois Gouget <fgouget@codeweavers.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 src/udscs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/udscs.c b/src/udscs.c
index 414dce5..fdd75a4 100644
--- a/src/udscs.c
+++ b/src/udscs.c
@@ -495,11 +495,12 @@ int udscs_server_fill_fds(struct udscs_server *server, fd_set *readfds,
         fd_set *writefds)
 {
     struct udscs_connection *conn;
-    int nfds = server->fd + 1;
+    int nfds;
 
     if (!server)
         return -1;
 
+    nfds = server->fd + 1;
     FD_SET(server->fd, readfds);
 
     conn = server->connections_head.next;
-- 
2.16.1

